include/types/MRSortModel.h

```diff
@@ -0,0 +1,72 @@
+#ifndef MRSORTMODEL_H
+#define MRSORTMODEL_H
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "AtomicMCDAObject.h"
+#include "Categories.h"
+#include "Criteria.h"
+#include "PerformanceTable.h"
+
+class MRSortModel : public AtomicMCDAObject {
+public:
+  /**
+   * MRSortModel standard constructor
+   *
+   * @param criteria Criteria object
+   * @param lambda threshold
+   * @param id optional name of the model
+   */
+  MRSortModel(Criteria &criteria, PerformanceTable &profiles, float lambda,
+              std::string id = "model");
+
+  /**
+   * MRSortModel generator constructor. This
+   * constructor initializes the profiles, lambda and criteria weights at
+   * random.
+   *
+   * @param ap alternative performance object
+   * @param id optional name of the model
+   */
+  MRSortModel(int n_cat, int n_crit, std::string id = "model");
+
+  /**
+   * MRSortModel constructor by copy
+   *
+   * @param mrsort MRSortModel object to copy
+   */
+  MRSortModel(const MRSortModel &mrsort);
+  ~MRSortModel();
+
+  friend std::ostream &operator<<(std::ostream &out, const MRSortModel &mrsort);
+
+  /**
+   * getId getter of id parameter
+   *
+   * @return id
+   */
+  std::string getId() const;
+
+  /**
+   * categoryAssignment assign the categories given the performance table
+   * and the current state of the model.
+   *
+   * @param pt PerformanceTable
+   *
+   * @return category_assignment pair vector with first value is the alternative
+   * id and second value the category assigned.
+   */
+  std::vector<std::pair<std::string, std::string>>
+  categoryAssignment(PerformanceTable &pt);
+
```

Comment:
We should probably use the type that we created. If will be easier to debug and we will have better code readability.
Either we should return an unordered_map (same as in AlternativePerformance) or and AlternativePerformance type.

Reply:
We should also make a function that takes as input one Performance type and send its assigned Category type

Reply:
Yes I agree, it would be better here to return an AlternativesPerformance, however I was kinda block on the matter: from a model perspective, I don't have access to the Category object in the current implementation. 
I will add the Categories object of the problem to the definition of the model (by reference) to fill the gap

Reply:
I could do the one with Performance -> Category if you think we will have use of it, but as a Performance table is not a vector<Performance> anymore, we won't be able to iterate over a PerformanceTable and use this new function.
 Let me know if you still think it will be useful to add it for the future

Reply:
You can also access the category via the nb of row of your profile PerformanceTable !
If you got 4 profiles that means that you have 5 classes so you can reconstruct the Category object and not pass it into parameter. 👍

Reply:
By doing that we take the risk to have different names between the initial categories and those ones as the initial ones could have been created with a prefix we don't know of... This might be a particular situation but it can be a breaking point so we will probably need to pass the Categories to model to be safe